File: app/src/main/java/com/kobrakid/retroachievements/Consts.java
package com.kobrakid.retroachievements;

@SuppressWarnings("SpellCheckingInspection")
public final class Consts {

    // Activity Request Codes
    public static final int BEGIN_LOGIN = 0;
    public static final int PULL_API_KEY = 1;
    // Activity Response Codes
    public static final int SUCCESS = 0;
    public static final int FAILURE = 1;
    public static final int CANCELLED = 2;

    // API
    public static final String BASE_URL = "https://retroachievements.org";
    public static final String API_URL = "API";
    public static final String FORUM_POSTFIX = "viewtopic.php?t=";
    public static final String GAME_POSTFIX = "game";
    public static final String GAME_BADGE_POSTFIX = "Badge";
    public static final String KEY_URL = "APIDemo.php";
    public static final String LEADERBOARDS_POSTFIX = "leaderboardList.php";
    public static final String LEADERBOARDS_INFO_POSTFIX = "leaderboardinfo.php?i=";
    public static final String LINKED_HASHES_POSTFIX = "linkedhashes.php?g=";
    public static final String USER_PIC_POSTFIX = "UserPic";
    public static final String USER_POSTFIX = "user";

    // Theming
    public static final String[] THEMES = new String[]{"RetroAchievements Themes", "Blank", "TwentySixteen", "Green", "Pony", "Red", "Spooky", "More Themes"};
    public static final boolean[] THEMES_ENABLE_ARRAY = new boolean[]{false, true, true, true, true, true, true, false};

    private Consts() {
        throw new AssertionError();
    }
}

File: app/src/main/java/com/kobrakid/retroachievements/Consts.kt
package com.kobrakid.retroachievements

class Consts private constructor() {
    companion object {
        // API
        const val BASE_URL = "https://retroachievements.org"
        const val API_URL = "API"
        const val FORUM_POSTFIX = "viewtopic.php?t="
        const val GAME_POSTFIX = "game"
        const val GAME_BADGE_POSTFIX = "Badge"
        const val KEY_URL = "APIDemo.php"
        const val LEADERBOARDS_POSTFIX = "leaderboardList.php"
        const val LEADERBOARDS_INFO_POSTFIX = "leaderboardinfo.php?i="
        const val LINKED_HASHES_POSTFIX = "linkedhashes.php?g="
        const val USER_PIC_POSTFIX = "UserPic"
        const val USER_POSTFIX = "user"
    }

    @Suppress("unused")
    enum class Theme(val themeName: String, val themeAttr: Int, val enabled: Boolean = true) {
        RA("RetroAchievements Themes", 0, false),
        BLANK("Blank", R.style.BlankTheme),
        TWENTYSIXTEEN("TwentySixteen", R.style.TwentySixteenTheme),
        GREEN("Green", R.style.GreenTheme),
        PONY("Pony", R.style.PonyTheme),
        RED("Red", R.style.RedTheme),
        SPOOKY("Spooky", R.style.SpookyTheme),
        MORE("More Themes", 0, false),
        SYSTEM("System Default", R.style.MyDayNight);

        override fun toString(): String {
            return themeName
        }
    }

    init {
        throw AssertionError()
    }
}